tribits/core/package_arch/TribitsAddOptionAndDefine.cmake

```diff
@@ -73,8 +83,8 @@ macro(tribits_add_option_and_define  USER_OPTION_NAME  MACRO_DEFINE_NAME
       global_set(${MACRO_DEFINE_NAME} OFF)
     endif()
   endif()
-  if (COMMAND tribits_pkg_export_cache_var)
-    tribits_pkg_export_cache_var(${USER_OPTION_NAME})
+  tribits_pkg_export_cache_var(${USER_OPTION_NAME})
+  if(NOT ${MACRO_DEFINE_NAME} STREQUAL "")
```

Comment:
Was this supposed to be:

```cmake
if(NOT MACRO_DEFINE_NAME STREQUAL "")
```

?

Reply:
That is supposed to work but I seem to have seen cases over the years where that failed.  The most robust check sesems to be:

```
if(NOT "${MACRO_DEFINE_NAME}" STREQUAL "")
```

Perhaps this has been fixed in newer versions of CMake.

Reply:
I addressed this in the commit https://github.com/TriBITSPub/TriBITS/commit/c5805daa1b2f8d04e7c00717dbca20b9e61133a9 that I directly pushed to 'master' (after testing locally).